UnchainedLauncherGUI/src/ViewModels/LauncherViewModel.cs

```diff
@@ -124,44 +124,35 @@ public void LaunchModded(Option<ServerLaunchOptions> serverOpts) {
-                                // rarer/harder to find by delegating the problem code
-                                // to a default
-                                PublicPorts ports = new(
-                                    Window.ServerSettingsViewModel.GamePort,
-                                    Window.ServerSettingsViewModel.PingPort,
-                                    Window.ServerSettingsViewModel.A2sPort
-                                );
-                                C2ServerInfo serverInfo = new() {
-                                    Ports = ports,
-                                    Name = Window.ServerSettingsViewModel.ServerName,
-                                    Description = Window.ServerSettingsViewModel.ServerDescription,
-                                    PasswordProtected = Window.ServerSettingsViewModel.ServerPassword.Length == 0,
-                                    Mods = ModManager.EnabledModReleases.Select(release =>
-                                        new ServerBrowserMod(
-                                            release.Manifest.Name,
-                                            release.Version.ToString(),
-                                            release.Manifest.Organization
-                                        )
-                                    ).ToArray()
-                                };
-
-                                Window.ServersViewModel.RegisterServer("127.0.0.1", ports, Window.ServerSettingsViewModel.RconPort, serverInfo, process);
+                                serverOpts.IfSome( serverOpts => {
+                                    PublicPorts ports = new(
+                                        serverOpts.GamePort,
+                                        serverOpts.BeaconPort,
+                                        serverOpts.QueryPort
+                                    );
+                                    C2ServerInfo serverInfo = new() {
+                                        Ports = ports,
+                                        Name = serverOpts.Name,
+                                        Description = serverOpts.Description,
+                                        PasswordProtected = serverOpts.Password.IsSome,
+                                        Mods = ModManager.EnabledModReleases.Select(release =>
+                                            new ServerBrowserMod(
+                                                release.Manifest.Name,
+                                                release.Version.ToString(),
+                                                release.Manifest.Organization
+                                            )
+                                        ).ToArray()
+                                    };
+
+                                    Window.ServersViewModel.RegisterServer("127.0.0.1", ports, serverOpts.RconPort, serverInfo, process);
```

Comment:
This line used to be a bit more complicated. I delegated the process of setting up the views and other dependent objects in to the servers view model via the new `RegisterServer function`.